packages/salesforcedx-utils-vscode/src/commands/commandletExecutors.ts

```diff
@@ -114,27 +114,33 @@ export abstract class SfdxCommandletExecutor<T>
 
 export abstract class LibraryCommandletExecutor<T>
   implements CommandletExecutor<T> {
+  private readonly executionName: string;
+  private readonly logName: string;
+  private readonly outputChannel: vscode.OutputChannel;
+  protected readonly telemetry = new TelemetryBuilder();
+
   /**
-   * Command name visible to user while executing.
-   */
-  protected abstract readonly executionName: string;
-  /**
-   * Command name for logging purposes such as telemetry
-   */
-  protected abstract readonly logName: string;
-  /**
-   * Output channel to report execution status to.
+   * @param name Name visible to user while executing.
+   * @param logName Name for logging purposes such as telemetry.
+   * @param outputChannel VS Code output channel to report execution status to.
    */
-  protected abstract readonly outputChannel: vscode.OutputChannel;
-  protected readonly telemetry = new TelemetryBuilder();
+  constructor(
+    executionName: string,
+    logName: string,
+    outputChannel: vscode.OutputChannel
+  ) {
+    this.executionName = executionName;
+    this.logName = logName;
+    this.outputChannel = outputChannel;
```

Comment:
Now provide the vs code output channel to report execution status with, since extensions could be using their own channels with the commandlet.

Reply:
Why did we choose to provide the OutputChannel instead of just the name of the channel ? We could do `ChannelService.getInstance(<provided channel name>)` instead of doing `new ChannelService(this.outputChannel)`

What happens if we try to create an output channel has already been created ?

Reply:
If it's done by name with `getInstance`, we run into the possibility of creating multiple channels with the same label. I was curious about that as well and tried it out. We need to pass the same channel object to commands in order for them to send output to the same window. An implication of this is as long as we want commands across different extensions to do that, the core extension must export it and the others have to add core as a dependency.